packages/nextjs/hooks/scaffold-eth/useAccountBalance.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from "react";
+import { useTargetNetwork } from "./useTargetNetwork";
+import { useQueryClient } from "@tanstack/react-query";
+import { UseBalanceParameters, useBalance, useBlockNumber } from "wagmi";
+
+type UseAccountBalanceParameters = UseBalanceParameters & { watch?: boolean };
+
+export const useAccountBalance = ({ watch = true, ...useBalanceParameters }: UseAccountBalanceParameters) => {
+  const { targetNetwork } = useTargetNetwork();
+  const queryClient = useQueryClient();
+  const { data: blockNumber } = useBlockNumber({ watch, chainId: targetNetwork.id });
+  const { queryKey, ...restUseBalanceReturn } = useBalance(useBalanceParameters);
+
+  useEffect(() => {
+    queryClient.invalidateQueries({ queryKey });
```

Comment:
maybe add a condition above this line and add `watch` as dependency to `useEffect` : 

```ts
useEffect(() => {
 if(!watch) return;
 queryClient.invalidateQueries({ queryKey });

 }, [blockNumber, watch]);
```

This is allow people to keep `watch` property dynamic(derived from a state maybe) in case they want. 

Should we take `enabled` in consideration too before calling `queryClient.invalidateQueries({ queryKey });`  ? Lol just thinking out loud without trying

Reply:
That `useEffect` was triggered only when `blockNumber` change, and `blockNumber` changed only `watch` was `true`, so looks like it's unnecessary.

But I removed watch param, so again we don't need that `if`

> Should we take enabled in consideration too before calling queryClient.invalidateQueries({ queryKey }); ? Lol just thinking out loud without trying

I don't know but tend to say no until we will need it often

Thanks for suggestions!

Reply:
> But I removed watch param, so again we don't need that if

Ohh yes yes now it is not needed at all 🙌